From 90db095108988932c2dfcacf637ddf5cf20d468e Mon Sep 17 00:00:00 2001
From: gollarx <ratnakumarix.golla@intel.com>
Date: Thu, 19 May 2022 18:30:31 +0530
Subject: [PATCH] Revert "Upgrade the qemu4.2 to 6.0 and add SRIOV changes"

This reverts commit 981fde7afd133b128e831695b9aa983dbc71b8dc.

Tracked-On: OAM-102241
---
 scripts/setup_host.sh      | 37 ++++++-------------------------------
 scripts/start_flash_usb.sh |  2 +-
 2 files changed, 7 insertions(+), 32 deletions(-)

diff --git a/scripts/setup_host.sh b/scripts/setup_host.sh
index a698c82..3ff0f40 100755
--- a/scripts/setup_host.sh
+++ b/scripts/setup_host.sh
@@ -9,7 +9,7 @@ set -eE
 
 #---------      Global variable     -------------------
 reboot_required=0
-QEMU_REL="qemu-6.0.0"
+QEMU_REL="qemu-4.2.0"
 CIV_WORK_DIR=$(pwd)
 CIV_GOP_DIR=$CIV_WORK_DIR/GOP_PKG
 CIV_VERTICAl_DIR=$CIV_WORK_DIR/vertical_patches/host
@@ -39,16 +39,15 @@ function ubu_install_qemu_gvt(){
     sudo apt purge -y "^qemu"
     sudo apt autoremove -y
     sudo apt install -y git libfdt-dev libpixman-1-dev libssl-dev vim socat libsdl2-dev libspice-server-dev autoconf libtool xtightvncviewer tightvncserver x11vnc uuid-runtime uuid uml-utilities bridge-utils python-dev liblzma-dev libc6-dev libegl1-mesa-dev libepoxy-dev libdrm-dev libgbm-dev libaio-dev libusb-1.0-0-dev libgtk-3-dev bison libcap-dev libattr1-dev flex libvirglrenderer-dev build-essential gettext libegl-mesa0 libegl-dev libglvnd-dev libgl1-mesa-dev libgl1-mesa-dev libgles2-mesa-dev libegl1 gcc g++ pkg-config libpulse-dev libgl1-mesa-dri
-    sudo apt install -y ninja-build libcap-ng-dev
 
     [ ! -f $CIV_WORK_DIR/$QEMU_REL.tar.xz ] && wget https://download.qemu.org/$QEMU_REL.tar.xz -P $CIV_WORK_DIR
     [ -d $CIV_WORK_DIR/$QEMU_REL ] && rm -rf $CIV_WORK_DIR/$QEMU_REL
     tar -xf $CIV_WORK_DIR/$QEMU_REL.tar.xz
 
     cd $CIV_WORK_DIR/$QEMU_REL/
+    patch -p1 < $CIV_WORK_DIR/patches/qemu/Disable-EDID-auto-generation-in-QEMU.patch
     patch -p1 < $CIV_WORK_DIR/patches/qemu/0001-Revert-Revert-vfio-pci-quirks.c-Disable-stolen-memor.patch
     patch -p1 < $CIV_WORK_DIR/patches/qemu/0002-qemu-change-fence-poll-time-by-current-workload.patch
-    patch -p1 < $CIV_WORK_DIR/patches/qemu/0003-Disable-EDID-auto-generation-in-QEMU.patch
     if [ -d $CIV_GOP_DIR ]; then
         for i in $CIV_GOP_DIR/qemu/*.patch; do patch -p1 < $i; done
     fi
@@ -93,7 +92,7 @@ function ubu_build_ovmf_gvt(){
     sudo apt install -y uuid-dev nasm acpidump iasl
     cd $CIV_WORK_DIR/$QEMU_REL/roms/edk2
 
-    patch -p1 < $CIV_WORK_DIR/patches/ovmf/0001-OvmfPkg-add-IgdAssignmentDxe.patch
+    patch -p4 < $CIV_WORK_DIR/patches/ovmf/OvmfPkg-add-IgdAssgingmentDxe-for-qemu-4_2_0.patch
     if [ -d $CIV_GOP_DIR ]; then
         for i in $CIV_GOP_DIR/ovmf/*.patch; do patch -p1 < $i; done
         cp $CIV_GOP_DIR/ovmf/Vbt.bin OvmfPkg/Vbt/Vbt.bin
@@ -117,7 +116,7 @@ function ubu_build_ovmf_gvt(){
 
     source ./edksetup.sh
     make -C BaseTools/
-    build -b DEBUG -t GCC5 -a X64 -p OvmfPkg/OvmfPkgX64.dsc -D NETWORK_IP4_ENABLE -D NETWORK_ENABLE  -D SECURE_BOOT_ENABLE -D TPM_ENABLE
+    build -b DEBUG -t GCC5 -a X64 -p OvmfPkg/OvmfPkgX64.dsc -D NETWORK_IP4_ENABLE -D NETWORK_ENABLE  -D SECURE_BOOT_ENABLE -DTPM2_ENABLE=TRUE
     cp Build/OvmfX64/DEBUG_GCC5/FV/OVMF.fd ../../../OVMF.fd
 
     if [ -d $CIV_GOP_DIR ]; then
@@ -184,28 +183,6 @@ function ubu_enable_host_gvt(){
     fi
 }
 
-function ubu_enable_host_sriov(){
-   if [[ ! `cat /etc/default/grub` =~ "i915.enable_guc=0x7" ]]; then
-        read -p "The grub entry in '/etc/default/grub' will be updated for enabling SRIOV, do you want to continue?     [Y/n]" res
-        if [ x$res = xn ]; then
-            exit 0
-        fi
-	sed -i "s/GRUB_CMDLINE_LINUX=\"/GRUB_CMDLINE_LINUX=\"i915.enable_guc=0x7 udmabuf.list_limit=8192  /g" /etc/default/grub
-        update-grub
-
-        echo -e "\nkvmgt\nvfio-iommu-type1\nvfio-mdev\n" >> /etc/initramfs-tools/modules
-        update-initramfs -u -k all
-
-        # Switch to Xorg for Ubuntu 21.04
-        if [[ $(lsb_release -rs) == "21.04" ]]; then
-            sed -i "s/\#WaylandEnable=false/WaylandEnable=false/g" /etc/gdm3/custom.conf
-        fi
-
-        reboot_required=1
-
-    fi
-}
-
 function check_os() {
     local version=`cat /proc/version`
 
@@ -328,7 +305,6 @@ function ubu_install_lg_client(){
 }
 
 function set_host_ui() {
-    setup_power_button
     if [[ $1 == "headless" ]]; then
         setup_power_button
         [[ $(systemctl get-default) == "multi-user.target" ]] && return 0
@@ -414,8 +390,8 @@ function ubu_update_bt_fw() {
         reboot_required=1
     else
         usb_devices="/sys/kernel/debug/usb/devices"
-        count="$(grep -c 'Cls=e0(wlcon) Sub=01 Prot=01 Driver=btusb' $usb_devices || true)"
-        if [ $count -eq 0 ]; then
+        cat $usb_devices | grep  -q "Cls=e0(wlcon) Sub=01 Prot=01 Driver=btusb"
+        if [ $? != 0 ]; then
             echo " Skip the host BT firmware update as BT controller is not present"
         else
             echo "Host Bluetooth firmware update failed. Run the setup again after cold reboot"
@@ -514,7 +490,6 @@ ubu_changes_require
 ubu_install_qemu_gvt
 ubu_build_ovmf_gvt
 ubu_enable_host_gvt
-ubu_enable_host_sriov
 
 install_vm_manager
 
diff --git a/scripts/start_flash_usb.sh b/scripts/start_flash_usb.sh
index df2f88d..7beb60d 100755
--- a/scripts/start_flash_usb.sh
+++ b/scripts/start_flash_usb.sh
@@ -87,7 +87,7 @@ qemu-system-x86_64 \
   -k en-us \
   -vga qxl \
   -machine kernel_irqchip=on \
-  -chardev socket,id=charserial0,path=./kernel-console,server=on,wait=off \
+  -chardev socket,id=charserial0,path=./kernel-console,server,nowait \
   -device isa-serial,chardev=charserial0,id=serial0 \
   -device qemu-xhci,id=xhci,addr=0x5 \
   -drive file=./flash.vfat,id=udisk1,format=raw,if=none \
-- 
2.36.1

